DSpace/dspace-angular
# Made legacy bitstream URLs redirect with 301 status code

## References
* Fixes #2963
* Fixes #2727

## Description
This fixes an issue where legacy bitstream URLs are redirected with status code 302 instead of 301 (permanent URL redirect).

## Instructions for Reviewers
List of changes in this PR:
* Refactored the existing legacy url resolver into a redirect guard and updated the status code of the request before redirecting to the

**Guidance for how to test & review this PR:**
- Run your frontend in production mode with sandbox.dspace.org as backend
- Run `curl --head http://localhost:4000/bitstream/handle/123456789/258/Money%20and%20Emerging%20Adults.pdf` and verify that the status code is now 301
- Run `curl --head http://localhost:4000/bitstream/handle/123456789/258/Money%20and%20Emerging` and verify that the status code for an invalid url is still 404

## Checklist
